github/resource_github_branch_test.go

```diff
@@ -0,0 +1,204 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-github/v29/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccGithubBranch_basic(t *testing.T) {
+
+	var (
+		reference github.Reference
+
+		name   = "basic"
+		repo   = "test-repo"
+		branch = "test-branch-" + acctest.RandString(5)
+		ref    = "refs/heads/" + branch
+		rn     = "github_branch." + name
+		id     = repo + ":" + branch
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGithubBranchDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGithubBranchConfig(name, repo, branch),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubBranchExists(rn, id, &reference),
+					resource.TestCheckResourceAttr(rn, "repository", repo),
+					resource.TestCheckResourceAttr(rn, "branch", branch),
+					resource.TestCheckResourceAttr(rn, "source_branch", "master"),
+					resource.TestCheckResourceAttrSet(rn, "source_sha"),
+					resource.TestCheckResourceAttrSet(rn, "etag"),
+					resource.TestCheckResourceAttr(rn, "ref", ref),
+					resource.TestCheckResourceAttrSet(rn, "sha"),
+				),
+			},
+			{
+				Config: testAccCheckGithubBranchConfig(name, repo, branch),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubBranchExists(rn, id, &reference),
+					resource.TestCheckResourceAttr(rn, "repository", repo),
+					resource.TestCheckResourceAttr(rn, "branch", branch),
+					resource.TestCheckResourceAttr(rn, "source_branch", "master"),
+					resource.TestCheckResourceAttrSet(rn, "source_sha"),
+					resource.TestCheckResourceAttrSet(rn, "etag"),
+					resource.TestCheckResourceAttr(rn, "ref", ref),
+					resource.TestCheckResourceAttrSet(rn, "sha"),
+				),
+			},
+			{
+				ResourceName:      rn,
+				ImportState:       true,
+				ImportStateId:     fmt.Sprintf("%s:%s", repo, branch),
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"source_sha",
```

Comment:
What does work around?

Reply:
`source_sha` is extremely difficult to determine at import time. I dug through github's api and git. It's non-trivial to find the parent commit, and even once you do, it's not guaranteed accurate. It my testing, I was getting 
```
testing.go:654: Step 2 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) {
        }
        
        
        (map[string]string) (len=1) {
         (string) (len=10) "source_sha": (string) (len=40) "18edb0e04930c3e4cff23fb837d3116898e0c6a6"
        }
```